### PR TITLE
Automatically update Homebrew Cask on release

### PR DESCRIPTION
**Pull request by mjcheetham**
On release, automatically update the Homebrew Cask in the microsoft/git Tap.

### CODE REVIEW

File: .github/workflows/release-homebrew.yaml [hunk 1]
```diff
@@ -1,34 +1,18 @@
...
-      uses: mjcheetham/update-homebrew@v1
+      uses: mjcheetham/update-homebrew@v1.1
       with:
         token: ${{ secrets.HOMEBREW_TOKEN }}
         tap: microsoft/git
         name: git-credential-manager-core
         type: cask
-        version: ${{ steps.version.outputs.result }}
+        releaseAsset: gcmcore-osx-(.*)\.pkg
+        alwaysUsePullRequest: true
```
Comment (mjcheetham) [inline .github/workflows/release-homebrew.yaml:18]: There's a question if we want to go ahead and push a commit to the `master` branch in the Tap, **_OR_** create a PR. Creating a PR gives us one last 'manual check' option.

Comment (derrickstolee) [reply .github/workflows/release-homebrew.yaml:18]: Let's start with a PR and see how that goes. Thanks!

If we find that this is stable and we don't want the extra manual step, then we can change this for a later release.

File: .github/workflows/release-homebrew.yaml [hunk 2]
```diff
@@ -1,34 +1,18 @@
 name: "release-homebrew"
 on:
   release:
-    types: [published]
+    types: [released]
```
Comment (derrickstolee) [inline .github/workflows/release-homebrew.yaml:4]: Does this trigger on `pre-release` to `release` instead of publishing a draft to `pre-release`?

Comment (mjcheetham) [reply .github/workflows/release-homebrew.yaml:4]: So right now, this change will mean the Homebrew PR only gets created when doing a `release`/not-a-draft/not-a-pre-release.

**Pull request merged**